hotshot-types/src/light_client.rs

```diff
@@ -278,7 +278,7 @@ impl<F: PrimeField> GenericPublicInput<F> {
 
 pub fn hash_bytes_to_field<F: RescueParameter>(bytes: &[u8]) -> Result<F, RescueError> {
     // make sure that `mod_order` won't happen.
-    let bytes_len = ((<F as PrimeField>::MODULUS_BIT_SIZE + 7) / 8 - 1) as usize;
+    let bytes_len = <F as PrimeField>::MODULUS_BIT_SIZE.div_ceil(8) as usize;
```

Comment:
this is wrong. we forget the `-1`. the `+7)/8` is the `div_ceil()`

to see why: this bytes_len is suppose to avoid `mod p` to happen. if `MODULUS_BIT_SIZE = 9`, `9.div_ceil()=2`, and if I give a 10 bits value which is still within 2 bytes, it will wrap around `modulus`. 

I'm fixing the `-1` in my PR.

Reply:
Thanks @alxiong but I suggest to make separate PRs for fixes.

Reply:
true. but i need to sync with main and resolve conflicts. ... already fixed locally. if changes are significant, i promise to leave for future PRs